onboarding/source/amb_light_service.c

```diff
@@ -34,20 +47,82 @@ static void lightServiceTask(void * pvParameters);
 obc_error_code_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here. Return error code if task/queue was not created successfully.
+    if (lightServiceTaskHandle == NULL) {
+        // Create controller task
+        lightServiceTaskHandle = xTaskCreateStatic( lightServiceTask,             /* Function that implements the task. */
+                                                    LIGHT_SERVICE_NAME,            /* Text name for the task. */
+                                                    LIGHT_SERVICE_STACK_SIZE,      /* Stack size in words, not bytes. */
+                                                    NULL,                       /* Parameter passed into the task. */
+                                                    LIGHT_SERVICE_PRIORITY,        /* Priority at which the task is created. */
+                                                    lightServiceTaskStack,        /* Array to use as the task's stack. */
+                                                    &lightServiceTaskBuffer);     /* Used to hold the task's data structure */
+    }
+
+    if (lightServiceTaskHandle == NULL)
+        return OBC_ERR_CODE_TASK_CREATION_FAILED;
+
+    if (lightServiceQueueHandle == NULL){
+        lightServiceQueueHandle = xQueueCreateStatic (LS_QUEUE_LENGTH,
+                                                      LS_ITEM_SIZE,
+                                                      lightServiceQueueStorage,
+                                                      &lightServiceQueue  );
+    }
+    
+    if(lightServiceQueueHandle == NULL)
+        return OBC_ERR_CODE_QUEUE_CREATION_FAILED;
 
+    return OBC_ERR_CODE_SUCCESS;
     /* USER CODE END */
 }
 
+
+uint16_t getLightSensorData(void)
+{
+    //Function taken from https://git.ti.com/cgit/hercules_examples/hercules_examples/tree/Launchpad/RM/RM46L8/Project_1/demoapp/source/adc_demos.c
+	adcData_t adcData;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion(adcREG1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetData(adcREG1, adcGROUP1, &adcData);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return ((&adcData)->value);
```

Comment:
`adcData` is a struct (not a pointer), so you can just return `adcData.value`. The `->` operator just dereferences the pointer and then applies the `.` operator. 

I.e. `(&adcData)->value` = `(*(&adcData)).value` = `adcData.value`

Reply:
Also, you can resolve the PR comment once you've fixed the issue.